crates/red_knot_module_resolver/src/path.rs

```diff
@@ -0,0 +1,1179 @@
+        match value {
+            ModuleResolutionPathBufInner::Extra(path) => ModuleResolutionPathRefInner::Extra(path),
+            ModuleResolutionPathBufInner::FirstParty(path) => {
+                ModuleResolutionPathRefInner::FirstParty(path)
+            }
+            ModuleResolutionPathBufInner::StandardLibrary(path) => {
+                ModuleResolutionPathRefInner::StandardLibrary(path)
+            }
+            ModuleResolutionPathBufInner::SitePackages(path) => {
+                ModuleResolutionPathRefInner::SitePackages(path)
+            }
+        }
+    }
+}
+
+impl<'a> From<&'a ModuleResolutionPathBuf> for ModuleResolutionPathRef<'a> {
+    fn from(value: &'a ModuleResolutionPathBuf) -> Self {
+        ModuleResolutionPathRef(ModuleResolutionPathRefInner::from(&value.0))
+    }
+}
+
+impl<'a> PartialEq<ModuleResolutionPathBuf> for ModuleResolutionPathRef<'a> {
+    fn eq(&self, other: &ModuleResolutionPathBuf) -> bool {
+        match (self.0, &other.0) {
+            (
+                ModuleResolutionPathRefInner::Extra(self_path),
+                ModuleResolutionPathBufInner::Extra(other_path),
+            )
+            | (
+                ModuleResolutionPathRefInner::FirstParty(self_path),
+                ModuleResolutionPathBufInner::FirstParty(other_path),
+            )
+            | (
+                ModuleResolutionPathRefInner::StandardLibrary(self_path),
+                ModuleResolutionPathBufInner::StandardLibrary(other_path),
+            )
+            | (
+                ModuleResolutionPathRefInner::SitePackages(self_path),
+                ModuleResolutionPathBufInner::SitePackages(other_path),
+            ) => *self_path == **other_path,
+            _ => false,
+        }
+    }
+}
+
+impl<'a> PartialEq<FileSystemPath> for ModuleResolutionPathRef<'a> {
+    fn eq(&self, other: &FileSystemPath) -> bool {
+        self.0.as_file_system_path() == other
+    }
+}
+
+impl<'a> PartialEq<ModuleResolutionPathRef<'a>> for FileSystemPath {
+    fn eq(&self, other: &ModuleResolutionPathRef<'a>) -> bool {
+        self == other.0.as_file_system_path()
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use insta::assert_debug_snapshot;
+
+    use crate::db::tests::{create_resolver_builder, TestCase, TestDb};
+    use crate::supported_py_version::SupportedPyVersion;
+
+    use super::*;
+
+    // Replace windows paths
+    static WINDOWS_PATH_FILTER: [(&str, &str); 1] = [(r"\\\\", "/")];
+
+    #[test]
+    fn constructor_rejects_non_pyi_stdlib_paths() {
+        assert!(ModuleResolutionPathBuf::standard_library("foo.py").is_none());
+        assert!(ModuleResolutionPathBuf::standard_library("foo/__init__.py").is_none());
+        assert!(ModuleResolutionPathBuf::standard_library("foo.py.pyi").is_none());
+    }
+
+    fn stdlib_path_test_case(path: &str) -> ModuleResolutionPathBuf {
+        ModuleResolutionPathBuf::standard_library(path).unwrap()
+    }
+
```

Comment:
IMO, the helper function obscures the test logic because the function name isn't telling me anything about what `stdlib_test_case` is doing. I have to jump to the definition to understand the test case. I would just inline the method or rename the method to `unwrap_standard_library(path: &str)`